File: 95/solution.go
package main /*{{{*/

import (
	"fmt"
)

const (
	e4  = int64(10000)
	e8  = e4 * e4
	mod = 1e9 + 7
) /*}}}*/

const e6 = 1000000

var num = [e6 + 10]int{}
var vis = [e6 + 10]bool{}

func cal_next(a int) int {
	ret := 1
	for i := 2; i*i <= a; i++ {
		if a%i == 0 {
			ret += i
			if a/i != i {
				ret += a / i
			}
		}
	}
	return ret
}

func cal_len(i, origin int, show bool) int {
	if i <= 2 || i > e6 {
		return -100
	}
	if vis[i] && i == origin {
		return 0
	}
	if vis[i] {
		return -100
	}
	if show {
		fmt.Printf("%d ", i)
	}
	ret := 1
	vis[i] = true
	ret += cal_len(num[i], origin, show)
	vis[i] = false
	return ret
}
func main() {
	for i := 2; i <= e6; i++ {
		num[i] = cal_next(i)
	}
	max_len, index := 0, 0
	for i := 2; i <= e6; i++ {
		len := cal_len(i, i, false)
		if len > 0 {
			cal_len(i, i, true)
			fmt.Println(": ", len)
		}
		if len > max_len {
			max_len = len
			index = i
		}
	}
	fmt.Println(max_len, index)
}

File: 95/solution.cpp
#include<iostream>/*{{{*/
#include<vector>
#include<set>
#include<math.h>
#include<map>
#include "../template/bigint.cpp"
const int e4 = 10000;
const int e8 = e4*e4;
const int e6 = e4*100;
const int mod = 1e9+7;
using namespace std;/*}}}*/

/*add{{{*/
template <typename T>
T base_add(T a, T b) {
	a %= mod;
	b %= mod;
	return (a+b)%mod;
}
template <typename T>
T add(const T &t) { return t; }
template <typename T, typename ... Args>
T add(const T &t, const Args&... args) { return base_add(t, add(args...)); }
/*}}}*/

/*{{{mul*/
template <typename T>
T base_mul(T a, T b) {
	a%=mod;
	b%=mod;
	T ret(0);
	while(b) {
		if(b&1) ret = (ret + a) % mod;
		b>>=1;
		a = (a+a)%mod;
	}
	return ret;
}
template <typename T>
T mul(const T &t) { return t; }
template <typename T, typename ... Args>
T mul(const T &t, const Args&... args) { return base_mul(t, mul(args...)); }
/*}}}*/

/*{{{qpow*/
template <typename T>
T qpow(T a, T b) {
	T ret(1);
	while(b) {
		if(b&1) ret = mul(ret, a);
		a = mul(a, a);
		b>>=1;
	}
	return ret;
}
/*}}}*/

int num[e6+10];
bool vis[e6+10];
int cal_next(int a) {
	int ret = 1;
	for(int i = 2;i*i<=a;i++) if(a%i==0) {
		ret += i;
		if(a/i != i) ret += a/i;
	}
	return ret;
}
int cal_len(int i, int origin, bool show = false) {
	if(i<=2 || i > e6) return -100;
	if(vis[i] && i == origin) return 0;
	if(vis[i]) return -100;
	
	if(show) cout<<i<<' ';

	int ret = 1;
	vis[i] = true;
	ret += cal_len(num[i], origin, show);
	vis[i] = false;
	return ret;
}

int main() {
	for(int i = 2;i<=e6;i++) num[i] = cal_next(i);
	int max_len = 0;
	int index = 0;
	for(int i = 2;i<=e6;i++) {
		int len = cal_len(i, i);
		if(len>0) {
			cal_len(i, i, true);
			cout<<": "<<len<<endl;
		}
		if(len > max_len) {
			max_len = len;
			index = i;
		}
	}
	cout<<max_len<<' '<<index<<endl;
	return 0;
}
